src/main/java/seedu/address/ui/UpcomingTuitionListPanel.java

```diff
@@ -0,0 +1,123 @@
+package seedu.address.ui;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.logging.Logger;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.session.Session;
+import seedu.address.model.student.Student;
+import seedu.address.model.tuition.Tuition;
+
+/**
+ * Panel containing the list of the upcoming tuition in 1 week.
+ */
+public class UpcomingTuitionListPanel extends UiPart<Region> {
+    private static final String FXML = "UpcomingTuitionListPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(UpcomingTuitionListPanel.class);
+
+    @FXML
+    private ListView<Tuition> upcomingTuitionListView;
+
+    /**
+     * Creates a {@code UpcomingTuitionListPanel} with the given {@code ObservableList}.
+     */
+    public UpcomingTuitionListPanel(ObservableList<Student> studentList) {
+        super(FXML);
+        ObservableList<Tuition> tuitionList = FXCollections.observableArrayList();
+        ObservableList<Tuition> finalTuitionList = FXCollections.observableArrayList();
+
+        populateTuitionList(studentList, tuitionList);
+        filterOneWeekTuitionSessions(tuitionList, finalTuitionList);
+        sortByDate(finalTuitionList);
+
+        studentList.addListener((ListChangeListener<Student>) change -> {
+            while (change.next()) {
+                tuitionList.clear();
+                finalTuitionList.clear();
+                populateTuitionList(studentList, tuitionList);
+                filterOneWeekTuitionSessions(tuitionList, finalTuitionList);
+                sortByDate(finalTuitionList);
+                populateUpcomingTuitionListView(finalTuitionList);
+            }
+        });
+
+        populateUpcomingTuitionListView(finalTuitionList);
+    }
+
+    /**
+     * Populate the upcomingTuitionListView with the given {@code ObservableList}
```

Comment:
Just some minor consistency issue, this could end with a . as per your other javadoc sentences. Using populates instead of populate here might be better according to the java coding standard.

Reply:
Got it, I will make the change. Thanks for spotting!